libraries/splunk_logs.rb

```diff
@@ -0,0 +1,114 @@
+# coding: UTF-8
+
+# Cookbook Name:: cerner_splunk
+# File Name:: splunk_logs.rb
+#
+# Libraries for configuring splunk's internal log settings.
+
+require 'chef/provider'
+require 'chef/resource'
+
+class Chef
+  class Resource
+    # Chef Resource for managing Splunk log config.
+    class SplunkLogs < Chef::Resource
+      provides :splunk_logs if respond_to?(:provides)
+
+      def initialize(name, run_context = nil)
+        super
+        @resource_name = :splunk_logs
+        @action = :create
+        @allowed_actions = %i[create]
+        @location = name
+        @contents = {}
+      end
+
+      def location(arg = nil)
+        set_or_return(:location, arg, kind_of: String)
+      end
+
+      def contents(arg = nil)
+        set_or_return(:contents, arg, kind_of: Hash)
+      end
+    end
+  end
+end
+
+require_relative 'conf'
+
+class Chef
+  class Provider
+    # Chef Provider for managing Splunk log config
+    class SplunkLogs < Chef::Provider
+      provides :splunk_logs if respond_to?(:provides)
+
+      def whyrun_supported?
+        true
+      end
+
+      def load_current_resource
+        @current_resource ||= Chef::Resource::SplunkLogs.new(new_resource.name)
+      end
+
+      def action_create
+        manage_file(new_resource.location, 'splunkd' => new_resource.contents)
```

Comment:
There are two stanzas in the log.cfg file, splunkd and python. Why not allow configuration of the python logging too?